Drop cflags that only exist in Chromium's clang fork.

Upstream clang rejects these outright, so every compile fails:
  clang++: error: unknown argument: '-fdiagnostics-show-inlining-chain'
  clang++: error: unknown argument: '-fno-lifetime-dse'
  clang++: error: unknown argument: '-fsanitize-ignore-for-ubsan-feature=...'

--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -590,10 +590,8 @@
     cflags += [ "-fcolor-diagnostics" ]
     if (!is_wasm) {
       if (!is_win) {
-        cflags += [ "-fdiagnostics-show-inlining-chain" ]
       } else {
         # Combine after https://github.com/llvm/llvm-project/pull/192241
-        cflags += [ "/clang:-fdiagnostics-show-inlining-chain" ]
       }
     }
     if (diagnostics_print_source_range_info && !is_win) {
@@ -630,7 +628,6 @@
     # https://crbug.com/484082200 for background and https://crrev.com/c/7593035
     # for discussion.
     if (!is_wasm) {
-      cflags += [ "-fno-lifetime-dse" ]
     }
 
     # TODO(hans): Remove this once Clang generates better optimized debug info
--- a/build/config/sanitizers/sanitizers.gni
+++ b/build/config/sanitizers/sanitizers.gni
@@ -549,7 +549,6 @@
         # be usable even in release builds, i.e. as widely as possible.
         # It's important not to have full-on UBSan workarounds activate
         # just because we built support for a specific sanitizer.
-        "-fsanitize-ignore-for-ubsan-feature=${invoker.sanitizer}",
       ]
       if (defined(invoker.cflags)) {
         cflags += invoker.cflags
